player/models.py

```diff
@@ -49,6 +54,11 @@ def validate_and_save(self, birth_date, favorite_ball_game):
         else:
             self.favorite_ball_game = favorite_ball_game
 
+        if profile_picture:
+            try:
+                self.profile_pic.save(profile_picture.name, profile_picture, save=False)
```

Comment:
I believe you mean that in addition to the try-catch block ?

Reply:
I believe that even though it is called 'image-field,' it will not fail to save a different file (not an image file). Hence, if a user attempts to upload a non-image file, the attempt will succeed (because the `try `will succeed) and we will save the non-picture file instead.

Furthermore, it is essential to thoroughly test all the error cases that this method throws. We should ensure that for each tuple provided in:

```
@pytest.mark.parametrize(
'birth_date, favorite_ball_game, profile_pic',

```
we receive the relevant error message.